core/src/HIP/Kokkos_HIP_IsXnack.cpp

```diff
@@ -16,74 +16,23 @@
 
 /*--------------------------------------------------------------------------*/
 
-#include <string>
 #include <string_view>
-#include <optional>
-#include <fstream>
-
-#ifdef __linux__
-#include <stdio.h>
-#include <sys/utsname.h>
-#endif
+#include <cstdlib>
 
 namespace {
 
-#ifdef __linux__
-// try to get `uname -r`. Returns an empty optional for any problem
-std::optional<std::string> uname_r() {
-  struct utsname buffer;
-  if (uname(&buffer) != 0) {
-    return std::nullopt;
-  }
-  return std::optional<std::string>{buffer.release};
-}
-#endif
-
 // returns true iff environment variable HSA_XNACK=1
 bool hsa_xnack_enabled_in_host_environment() {
   const char* var = std::getenv("HSA_XNACK");
   return var && std::string_view{var} == "1";
 }
 
-// return true iff `CONFIG_HMM_MIRROR=y` is definitely in /boot/config-$(uname
-// -r) returns false for non-linux platforms or any other problem
-bool config_hmm_mirror_in_boot_config() {
-#ifdef __linux__
-  // figure out the boot config file name
-  std::optional<std::string> unameR = uname_r();
-  if (!unameR) {
-    // couldn't figure out linux release name
-    return false;
-  }
-  std::string bootConfigPath = std::string("/boot/config-") + unameR.value();
-
-  std::ifstream file(bootConfigPath);
-  if (!file.is_open()) {
-    // couldn't open file for whatever reason
-    return false;
-  }
-
-  std::string line;
-  while (std::getline(file, line)) {
-    if (line.find("CONFIG_HMM_MIRROR=y") != std::string::npos) {
-      return true;  // Found the string
-    }
-  }
-  return false;
-#else   // __linux__
-  return false;
-#endif  // __linux__
-}
```

Comment:
If you intent to just add it back, I would leave it.

Reply:
reverted in https://github.com/kokkos/kokkos/pull/7765/commits/0316c911c7e35f0f1e83e712955000d0da7e3f23